tests/net/mod.rs

```diff
@@ -914,7 +923,14 @@ where
         self.adversary = adv;
 
         // Step 1: Pick a message from the queue and deliver it; returns `None` if queue is empty.
-        let msg = self.messages.pop_front()?;
+        // We use a do-while loop to remove the prefix of messages addressed to removed nodes.
+        let mut msg;
+        while {
```

Comment:
I appreciate the clever construction, however I do share Steve's concerns here: https://github.com/rust-lang/rfcs/issues/1313#issuecomment-146671501

Especially since we already have a comment explaining the whole construct. I think `loop { ...; cond(); }`-style might be better here?

Reply:
Now I started thinking that the messages should be removed earlier, at dispatch time. That way we can guarantee that no messages to the removed node arrive to it when it is finally readded. That would have been an undesirable side-effect.